docs/other-topics/transactions.md

```diff
@@ -222,6 +222,84 @@ const sequelize = new Sequelize('sqlite::memory:', {
 
 **Note for MSSQL:** _The `SET ISOLATION LEVEL` queries are not logged since the specified `isolationLevel` is passed directly to `tedious`._
 
+## Manually retrying failed transactions
+
+Sequelize does not automically retries on potential write conflicts or deadlocks, when using a certain level of isolations, it is expected that transactions may fail due to potential write conflicts in concurrent transactions or deadlocks.
+
+You can programmatically fix this by relying on options from the [retry-as-promised](https://github.com/mickhansen/retry-as-promised/blob/master/README.md) library:
+
+```typescript
+const { Sequelize } = require('@sequelize/core');
+
+const sequelize = new Sequelize(process.env.DB_NAME, 
+    process.env.DB_NAME, 
+    process.env.DB_PASS, 
+    {
+    host: process.env.DB_HOST,
+    dialect: 'mysql',
+    // Here:
+    retry: {
+        max: 3, // this may cause memory leaks if you set the retries limit too much.
+        match: [
+          Sequelize.ConnectionError,
+          Sequelize.ConnectionRefusedError,
+          Sequelize.ConnectionTimedOutError,
+          Sequelize.OptimisticLockError,
+          Sequelize.TimeoutError,
+          /Deadlock/i, // retry-as-promised also takes RegExp, for example to find deadlocks error.
+        ],
+    }
+});
+```
+
+You can use other possible error classes [here](https://sequelize.org/api/v7/classes/error).
+
+List of the default retry options and the value when you passing `{}` to the retry option:
+
+```js
+{
+  retry: {
+    $current: 1,
```

Comment:
Probably don't want to document `$current` since it's an internal "private" variable for `retry-as-promised`

Reply:
> Probably don't want to document `$current` since it's an internal "private" variable for `retry-as-promised`

Makes sense